lib/index.js

```diff
@@ -24,6 +24,8 @@ function filterKey(key) {
     return !KEY_BLACKLIST.has(key) && key[0] !== "_";
 }
 
+
+/* eslint-disable jsdoc/valid-types -- doesn't allow `readonly`. TODO: check why */
```

Comment:
For some reason, `readonly string[]` is considered invalid by `jsdoc/valid-types`. I disabled the rule here until we figure out what's the problem because changing this type would affect the types this package publishes.

Reply:
Maybe worth opening an issue on the plugin?

Reply:
Good idea. The issue template suggested opening an issue on the parser, here it is:

https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/164

Reply:
It seems like https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/164 won't be fixed anytime soon since there's still no response.